src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -580,6 +580,10 @@ private void LookupForMatchingGitHubRepository(BuildData mergedBuild)
                 {
                     string[] segments = mergedBuild.AzureDevOpsRepository.Split('/');
                     string repoName = segments[segments.Length - 1];
+                    if (repoName.Contains("-Trusted"))
```

Comment:
Wondering if we should narrow this down somewhat. If there is ever a repo called "dotnet/trusted-something" that would end up mirrored as dotnet-trusted-something and this would do the wrong thing.

Do the -Trusted repos only exist in devdiv? If so maybe we can restrict it to:

* it's the devdiv Azdo instance 
* the repo ends with -Trusted
* only replace that last bit, not every instance of the string

Some tests would be good to make sure all sorts of repos do the right thing here.

Reply:
Yes -Trusted Repos only exists in devdiv. We only care about the repo name, also most of the devdiv repos uses github url 
eg : MS-Build uses github url.

```
Repository:    https://github.com/dotnet/msbuild
Branch:        bevillal/exp/revert-16.10-cpp
Commit:        133eb6586ac2af6151ddcf45fc6e9a1f7c7e18e0
Build Number:  20210129.4
Date Produced: 1/29/2021 11:03 AM
Build Link:    https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4422691
BAR Build Id:  77576
Released:      False
Channels:
```

For now only Nuget is using Azdo mirror to build. Msbuild still uses github to build. So I think we should be good with the change. 

I did a search of all the dnceng Azdo urls
1. Aspnet https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-aspnetcore
2. Runtime https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-runtime

Reply:
Yes, this change as is works for nuget, but it breaks the world for any repo in the future that has the word trusted anywhere in it.

https://github.com/dotnet/trusted-packages would be broken as it would be mirrored as https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-trusted-packages for example.

Reply:
Yeah I only thought of existing repos. Thanks for mentioning awesome edge cases!  I will make sure I incorporate that too..